src/main/java/de/hysky/skyblocker/skyblock/item/slottext/adders/PotionLevelAdder.java

```diff
@@ -1,26 +1,26 @@
 package de.hysky.skyblocker.skyblock.item.slottext.adders;
 
 import de.hysky.skyblocker.skyblock.item.slottext.SlotText;
-import de.hysky.skyblocker.skyblock.item.slottext.SlotTextAdder;
+import de.hysky.skyblocker.skyblock.item.slottext.SimpleSlotTextAdder;
 import de.hysky.skyblocker.utils.ItemUtils;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NbtCompound;
 import net.minecraft.nbt.NbtElement;
 import net.minecraft.screen.slot.Slot;
 import net.minecraft.text.Text;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 import java.util.List;
 
-public class PotionLevelAdder extends SlotTextAdder {
+public class PotionLevelAdder extends SimpleSlotTextAdder {
     @Override
-    public @NotNull List<SlotText> getText(Slot slot) {
-        final ItemStack stack = slot.getStack();
+    public @NotNull List<SlotText> getText(@Nullable Slot slot, @NotNull ItemStack stack, int slotId) {
         NbtCompound customData = ItemUtils.getCustomData(stack);
         String title = stack.getName().getString();
         if (customData.contains("potion_level", NbtElement.INT_TYPE) && !title.contains("Healer Class") && !title.contains("Class Passives")) {
             if (title.contains("Healer Level ")){
-                String level = title.replaceAll("[^0-9]", "");
+                String level = title.replaceAll("\\D", "");
```

Comment:
Can tell you're the average regex enjoyer.

Reply:
Ehehe